### ISSUE TITLE
Navbar isn't accessible - accessibility broken on moving from bootstrap to shiny because invalid HTML generated

### ISSUE THREAD

**Issue by RosalynLP** [rstudio/shiny#3820]
Hi, I have noticed a problem with the accessibility of navbars (specifically, the ability to navigate using only a keyboard and no mouse) created by `navbarPage` in shiny apps. As a minimal example consider the `app.R`

```r
library(shiny)

# Define UI for application that draws a histogram
ui <- navbarPage("App Title",
                 tabPanel("Plot"),
                 tabPanel("Summary"),
                 tabPanel("Table")
)

# Define server logic required to draw a histogram
server <- function(input, output) {

}

# Run the application
shinyApp(ui = ui, server = server)
```
which produces the html 
```html
<nav class="navbar navbar-default navbar-static-top" role="navigation">
  <div class="container-fluid">
    <div class="navbar-header">
      <span class="navbar-brand">App Title</span>
    </div>
    <ul class="nav navbar-nav" data-tabsetid="8029">
      <li class="active">
        <a href="#tab-8029-1" data-toggle="tab" data-value="Plot">Plot</a>
      </li>
      <li>
        <a href="#tab-8029-2" data-toggle="tab" data-value="Summary">Summary</a>
      </li>
      <li>
        <a href="#tab-8029-3" data-toggle="tab" data-value="Table">Table</a>
      </li>
    </ul>
  </div>
</nav>
<div class="container-fluid">
  <div class="tab-content" data-tabsetid="8029">
    <div class="tab-pane active" data-value="Plot" id="tab-8029-1"></div>
    <div class="tab-pane" data-value="Summary" id="tab-8029-2"></div>
    <div class="tab-pane" data-value="Table" id="tab-8029-3"></div>
  </div>
</div>
```
When using the Tab button to navigate through the nav bar you are initially able to tab through each tab in order and select any one. However, once a tab has previously been selected this is no longer available when tabbing through the navbar. 

The reason for this is that when a tab is selected the `class="active"` tag is added to the corresponding `<li>` element, as in the Plot tab in the example html above. You can see this in the bootstrap docs [here](https://getbootstrap.com/docs/4.0/components/navbar/#nav) . When the tab is then navigated away from the "active" class is removed. For the shiny app this leaves just `<li class>` , rather than returning it to `<li>`, as it was before. This is now invalid html and the item is skipped when tabbing through the navbar from now on. 

This doesn't happen in bootstrap because all \<li> items also have the "nav-item" class, so they are left as `<li class="nav-item">` afterwards. In shiny, however, this class is removed at [line 430 in bootstrap.R](https://github.com/rstudio/shiny/blob/62bb21d5b6eb34a5701f849c78a04ff5efd5b46d/R/bootstrap.R#LL430C1-L430C1). This has the effect that invalid html is being created, and accessibility is broken.

The same happens with items in dropdowns within the navbar.

Would you please be able to fix this problem by including a different class for all these elements, even if you don't want the nav-item class to be included?

**Comment by cpsievert** [comment]
Interesting, thanks. `navbarPage()` will soon be (officially) superseded by `bslib::page_navbar()`, seems doesn't have the same issue (but `<-`/`->` need to be used to navigate between different tabs) 

```r
library(shiny)

ui <- page_navbar(
  title = "App Title",
  nav("Plot", "plot"),
  nav("Summary", "summary"),
  nav("Table", "table")
)

shinyApp(ui, function(...) {})
```

**Comment by RosalynLP** [comment]
Great will try that, thanks for your swift reply

**Comment by robgGS** [comment]
Hi, I'm also experiencing the same issue however, my code uses fluidPage instead of navbarPage, as well as tabsetPanel, and tabPanel. Is there an alternative to switching to page_navbar?

**Comment by cpsievert** [comment]
Yes, use `bslib::page_fluid()` (instead of `fluidPage()`) and [`navset_tab()`](https://rstudio.github.io/bslib/reference/navset.html#navset-tab-)/`nav_panel()`

**Comment by RosalynLP** [comment]
@cpsievert Thanks for your help, I think this issue can be closed if you're happy

**Comment by robgGS** [comment]
@cpsievert Thanks for that. Do navset_tab and nav_panel come from bslib? Or Shiny? I can use page_fluid but not the other two.
